mypy/messages.py

```diff
@@ -1193,18 +1193,30 @@ def need_annotation_for_var(self, node: SymbolNode, context: Context,
                                 python_version: Optional[Tuple[int, int]] = None) -> None:
         hint = ''
         has_variable_annotations = not python_version or python_version >= (3, 6)
+        pep604_supported = python_version and python_version >= (3, 10)
```

Comment:
nit: feels like this should mirror the check for has_variable_annotations when python_version is None